readium/navigator/src/main/java/org/readium/r2/navigator/epub/css/ReadiumCss.kt

```diff
@@ -162,6 +161,9 @@ data class ReadiumCss(
             Layout.HtmlDir.Rtl -> "rtl"
         } ?: return
 
+        // Removes any dir attributes in html/body.
+        content.replace(0, content.length, content.replace(dirRegex, "$1"))
+
```

Comment:
Are you sure about that? It goes against the EPUB spec, and doesn't match Readium CSS recommendations either, as far as I could see.

Reply:
Yeah because if an HTML tag has two `dir` attributes, the web view displays an error.

Reply:
Sure, I meant that Readium CSS doesn't recommend to do anything when the direction is LTR. You're preventing a use case with RTL in some resources from working, without any benefit. I wonder if we shouldn't have a setting "Force language and direction in each resource".

Reply:
If we don't explicitly override any `dir` attribute, we can't set the reading progression of a book declared as RTL to LTR. Besides, the reader would break if not all resources use the same `dir`, as the global reading progression implies that it flows in a single direction across resources.

> You're preventing a use case with RTL in some resources from working, without any benefit.

Which use case?

> I wonder if we shouldn't have a setting "Force language and direction in each resource".

The `readingProgression` and `language` settings are just that. When you don't want to force it, you reset them to `null`.

Reply:
> Besides, the reader would break if not all resources use the same dir, as the global reading progression implies that it flows in a single direction across resources.

Good point, EPUB might be a bit too optimistic on this.

> The readingProgression and language settings are just that. When you don't want to force it, you reset them to null.

As I understand it, `null` means automatic behavior depending on publication metadata, it does not mean "no action".

Reply:
Multi-reading progression publications are supported by EPUB but we decided to ignore it in Readium as it breaks affordances across resources. Which means we need to override the `dir` attributes in all cases.